src/hooks/useTodos.ts

```diff
@@ -0,0 +1,44 @@
+import { useState } from 'react';
+import { Filter } from '../App';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../app/store';
+import { filterSlice } from '../features/filter';
+
+export const useTodos = () => {
+  const dispatch = useDispatch();
+  const [isTodosLoaded, setIsTodosLoaded] = useState(false);
+
+  const [query, setQuery] = useState('');
+
+  const handleResetQuery = () => {
+    setQuery('');
+    dispatch(filterSlice.actions.settingQuery(''));
+  };
+
+  const anotherTodos = useSelector((state: RootState) => state.todos);
+  const filterRightNow = useSelector(
+    (state: RootState) => state.filter.status as Filter,
```

Comment:
The `filterRightNow` selector casts the filter status to the `Filter` enum. Ensure that the filter status in the Redux state matches the values defined in the `Filter` enum to prevent any type mismatches.